crates/turborepo-lib/src/cli.rs

```diff
@@ -371,6 +388,33 @@ pub fn run(repo_state: Option<RepoState>) -> Result<Payload> {
         }
     };
 
+    // If this is a run command, and we know the actual invocation path, set the
+    // inference root, as long as the user hasn't overridden the cwd
+    if clap_args.cwd.is_none() {
+        if let Some(Command::Run(run_args)) = &mut clap_args.command {
+            if let Ok(invocation_dir) = env::var(INVOCATION_DIR_ENV_VAR) {
+                let invocation_path = Path::new(&invocation_dir);
+
+                // If repo state doesn't exist, we're either local turbo running at the root
+                // (current_dir), or inference failed If repo state does exist,
+                // we're global turbo, and want to calculate package inference based on the repo
+                // root
+                let this_dir = current_dir()?;
+                let repo_root = repo_state.as_ref().map(|r| &r.root).unwrap_or(&this_dir);
+                if let Ok(relative_path) = invocation_path.strip_prefix(repo_root) {
+                    debug!(
+                        "pkg_inference_root set to \"{}\"",
+                        relative_path.to_string_lossy()
+                    );
+                    run_args.pkg_inference_root = Some(relative_path.to_string_lossy().to_string());
```

Comment:
I think we should error if we can't convert the path to a valid UTF8 string since a lossy root path will just cause issues down stream.

Reply:
Good idea.